### PR TITLE
WIP WebSearch: optional Obelix integration

### PR DESCRIPTION
**Pull request by ddaze**
### TODO
- [ ]  Release obelix-client and update requirements.txt
### Commit Message
- Connects Invenio with the obelix-client Python module to Obelix
- Introduces the new Obelix Recommendation System, optionally
  let users rank results by recommendations built on
  their search, view and download history
- The Obelix integration require redis to work, if the
  redis server is down or if we are unable to connect,
  Invenio will fall back to sort the records latest first
- Introduces CFG_WEBSEARCH_OBELIX_REDIS and
  CFG_WEBSEARCH_OBELIX_USER_KEY config used to set which
  redis host Invenio should connect to in order to use the
  recommendations from Obelix
- Adds bibrank config file for use with Obelix
  (template_read_recommendations.cfg)
- Fetches recommendations generated by Obelix from redis
- Logs search results to redis
- Logs page views to redis
- Logs downloads to redis

Signed-off-by: David Zerulla ddaze@outlook.de
Co-authored-by: Fredrik Nygård Carlsen fredrik@carlsen.io

### CODE REVIEW

File: modules/bibdocfile/lib/bibdocfile_webinterface.py [hunk 1]
```diff
@@ -164,6 +165,15 @@ def getfile(req, form):
             else:
                 docname = args['docname']
 
+            if CFG_WEBSEARCH_OBELIX_REDIS:
+                try:
+                    obelix = get_obelix()
+                    obelix.log('download_after_search',
+                               user_info,
+                               self.recid)
+                except:
```
Comment (dset0x) [inline modules/bibdocfile/lib/bibdocfile_webinterface.py]: Bare `except`s are very rarely a good idea. Take a look [here](https://docs.python.org/2/howto/doanddont.html#except), especially the last paragraph.

**Comment by ludmilamarian** [comment]
I still see the old commit message, can you please amend them as discussed IRL?

**Comment by ludmilamarian** [comment]
@ddaze is this PR still relevant or the commits will be part of a new PR including the new developments?

**Comment by ddaze** [comment]
Its still relevant, I will rebase it to only contain the Obelix integration including the improvements from
CDS.
For the recommendations i will create a new separate PR.

**Comment by ddaze** [comment]
@ludmilamarian I updated the commit with all changes made on cds.
Can you please check that I didn't forget something?

**Pull request closed without merging**